source/preprocessor.tex

```diff
@@ -2102,7 +2102,7 @@
 \pnum
 A preprocessing directive of the form
 \begin{ncsimplebnf}
-\terminal{\# line} digit-sequence \terminal{"} \opt{s-char-sequence} \terminal{"} new-line
+\terminal{\# line} digit-sequence \terminal{"} \opt{string-literal} \terminal{"} new-line
```

Comment:
No.

First, a _string-literal_ includes the delimiting quotation marks, so this is a bad change.
Second, a _string-literal_ allows for an _encoding-prefix_ or rawness, but the status quo "form" match does not.  So, this is a bad change, because it would appear to allow such prefixes (and we don't know what they mean in this context).  A reference to "the character string literal" does not establish a suitable constraint.

Yes, p1 talks about a _string-literal_ with zero context, and that's a bug that needs to be fixed, but this intro sentence is at issue, not the rest of the specification without further elaboration.  What does C23 say in that area?